third_party/terraform/website/docs/r/bigquery_table.html.markdown

```diff
@@ -64,18 +97,94 @@ The following arguments are supported:
 
 * `labels` - (Optional) A mapping of labels to assign to the resource.
 
-* `schema` - (Optional) A JSON schema for the table.
+* `schema` - (Optional) A JSON schema for the table. For more information
+    see the [BigQuery API documentation](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#resource).
 
 * `time_partitioning` - (Optional) If specified, configures time-based
     partitioning for this table. Structure is documented below.
 
 * `view` - (Optional) If specified, configures this table as a view.
     Structure is documented below.
 
+The `external_data_configuration` block supports:
+
+* `autodetect` - (Required) - Let BigQuery try to autodetect the schema
+    and format of the table.
+
+* `compression` (Optional) - The compression type of the data source.
+    Valid values are "NONE" or "GZIP".
+
+* `cvs_options` (Optional) - Additional properties to set if
+    `source_format` is set to "CSV". Structure is documented below.
+
+* `google_sheets_options` (Optional) - Additional options if
+    `source_format` is set to "GOOGLE_SHEETS". Structure is
+    documented below.
+
+* `ignore_unknown_values` (Optional) - Indicates if BigQuery should
+    allow extra values that are not represented in the table schema.
+    If true, the extra values are ignored. If false, records with
+    extra columns are treated as bad records, and if there are too
+    many bad records, an invalid error is returned in the job result.
+    The default value is false.
+
+* `max_bad_records` (Optional) - The maximum number of bad records that
+    BigQuery can ignore when reading data.
+
+* `schema` (Optional) - A JSON schema for the data. Schema is required
+    for CSV and JSON formats schema is disallowed for Google Cloud
+    Bigtable, Cloud Datastore backups, and Avro formats. For more
+    information see the
+    [BigQuery API documentation](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#resource).
+
+* `schema_format` (Required) - The data format. Supported values are:
+    "CVS", "GOOGLE_SHEETS", "NEWLINE_DELIMITED_JSON", "AVRO",
+    and "DATSTORE_BACKUP". To use "GOOGLE_SHEETS"
+    the `scopes` must include
+    "https://www.googleapis.com/auth/drive.readonly".
+
+* `source_uris` - (Required) A list of the fully-qualified URIs that point to
+    your data in Google Cloud.
+
+The `cvs_options` block supports:
+
+* `allow_jagged_rows` (Optional) - Indicates if BigQuery should accept rows
+    that are missing trailing optional columns.
+
+* `allow_quoted_newlines` (Optional) - Indicates if BigQuery should allow
+    quoted data sections that contain newline characters in a CSV file.
+    The default value is false.
+
+* `encoding` (Optional) - The character encoding of the data. The supported
+    values are UTF-8 or ISO-8859-1.
+
+* `field_delimiter` (Optional) - The separator for fields in a CSV file.
+
+* `quote` (Optional) - The value that is used to quote data sections in a
```

Comment:
> If your data does not contain quoted sections, set the property value to an empty string. If your data contains quoted newline characters, you must also set the allowQuotedNewlines property to true. 

These seem worth including in the docs too?

Reply:
👌 Added in c4c1799